web/client/utils/SecurityUtils.js

```diff
@@ -6,18 +6,149 @@
+     * returned if the attribute could not be found.
+     */
+    findUserAttribute: function(attributeName) {
+        // getting the user attributes
+        let userAttributes = this.getUserAttributes();
+        if (!userAttributes || !attributeName ) {
+            // the user as no attributes or the provided attribute name is undefined
+            return undefined;
+        }
+        return userAttributes.find(attribute => attribute.name
+            && attribute.name.toLowerCase() === attributeName.toLowerCase());
+    },
+
+    /**
+     * Search in the user attributes an attribute that matchs the provided
+     * attribute name. The search will not be case sensitive. Undefined is
+     * returned if the attribute could not be found otherwise the attribute
+     * value is returned.
+     */
+    findUserAttributeValue: function(attributeName) {
+        let userAttribute = this.findUserAttribute(attributeName);
+        return userAttribute && userAttribute.value;
+    },
+
+    /**
+     * Returns an array with the configured authentication rules. If no rules
+     * were configured an empty array is returned.
+     */
+    getAuthenticationRules: function() {
+        return ConfigUtils.getConfigProp('authenticationRules') || [];
+    },
+
+    /**
+     * Checks if authentication is activated or not.
+     */
+    isAuthenticationActivated: function() {
+        return ConfigUtils.getConfigProp('useAuthenticationRules') || false;
+    },
+
+    /**
+     * Returns the authentication method that should be used for the provided URL.
+     * We go through the authentication rules and find the first one that matchs
+     * the provided URL, if no rule matchs the provided URL undefined is returned.
+     */
+    getAuthenticationMethod: function(url) {
+        const foundRule = this.getAuthenticationRules().find(
+            rule => rule && rule.urlPattern && url.match(new RegExp(rule.urlPattern, "i")));
+        return foundRule ? foundRule.method : undefined;
+    },
+
+    /**
+     * This method will add query parameter based authentications to an url.
+     */
+    addAuthenticationToUrl: function(url) {
+        if (!url || !this.isAuthenticationActivated()) {
+            return url;
+        }
+        const parsedUrl = URL.parse(url, true);
+        parsedUrl.query = this.addAuthenticationParameter(url, parsedUrl.query);
+        // we need to remove this to force the use of query
+        delete parsedUrl.search;
+        return URL.format(parsedUrl);
+    },
+
+    /**
+     * This method will add query parameter based authentications to an object
+     * containing query paramaters.
+     */
+    addAuthenticationParameter: function(url, parameters) {
+        if (!url || !this.isAuthenticationActivated()) {
+            return parameters;
+        }
+        switch (this.getAuthenticationMethod(url)) {
+            case 'authkey':
+                const userUuid = this.findUserAttributeValue('uuid');
```

Comment:
token is already present in the security object and is more general (another login service should provide a temporary token that have to be used during the session instead of the static authkey. 
Use security.token.